UnitystationLauncher/Services/GoodFileService.cs

```diff
@@ -0,0 +1,116 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Net.Http;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Serilog;
+using UnitystationLauncher.Infrastructure;
+using UnitystationLauncher.Models.Enums;
+using UnitystationLauncher.Services.Interface;
+
+namespace UnitystationLauncher.Services;
+
+public class GoodFileService : IGoodFileService
+{
+    private readonly HttpClient _httpClient;
+    
+    private readonly IPreferencesService _preferencesService;
+    private readonly IEnvironmentService _environmentService;
+
+    private const string GoodFileURL = "Https://unitystationfile.b-cdn.net/GoodFiles/";
+    
+    public GoodFileService(HttpClient httpClient, IPreferencesService preferencesService, IEnvironmentService environmentService)
+    {
+        _httpClient = httpClient;
+        _preferencesService = preferencesService;
+        _environmentService = environmentService;
+
+    }
+
+    public async Task<(string, bool)> GetGoodFileVersion(string version)
+    {
+        if (await ValidGoodFilesVersion(version) == false)
+        {
+            return ("", false);
+        }
+  
+        var pathBase = _preferencesService.GetPreferences().InstallationPath;
+        var folderName = GetFolderName(version);
+        var versionPath = Path.Combine(pathBase, version, folderName);
+        
+        if (Directory.Exists(versionPath) == false)
+        {
+            
+            HttpResponseMessage request = await _httpClient.GetAsync(GoodFileURL + version +"/" + folderName + ".zip", HttpCompletionOption.ResponseHeadersRead);
+            await using Stream responseStream = await request.Content.ReadAsStreamAsync();
+            ZipArchive archive = new(responseStream);
+            archive.ExtractToDirectory(Path.Combine(versionPath), true);
+        }
+        
+        return (versionPath, true);
+    }
+
+    private string GetFolderName(string version)
+    {
+        var OS = _environmentService.GetCurrentEnvironment();
+        switch (OS)
+        {
+            case CurrentEnvironment.WindowsStandalone:
+                return version + "_Windows";
+            case CurrentEnvironment.LinuxStandalone:
+                return version + "_Linux";
+            case CurrentEnvironment.MacOsStandalone:
+                return version + "_Mac";
+        }
+
+        return "idk";
```

Comment:
`CurrentEnvironment.LinuxFlatpak` will return 'idk' here, I think it should be safe to use the same as the `CurrentEnvironment.LinuxStandalone`.